source/read/retrieve.txt

```diff
@@ -0,0 +1,229 @@
+If more than one document matches the query
+filter, this method returns the *first* matching document from the retrieved
+results as a Python dictionary. If no documents match the query filter, the method returns
+``None``.
+
+.. tip::
+   
+   The ``find_one()`` method is useful when you know there's only one matching document,
+   or you're only interested in the first match.  
+
+The following examples use the ``find_one()`` method to find the first document where
+the ``"cuisine"`` field has the value ``"Bakery"``. Select the corresponding tab to see
+an example in the Python shell or a ``.py`` file. 
+
+.. tabs::
+
+   .. tab:: File
+      :tabid: find-one-file
+
+      .. code-block:: python
+         :copyable: true
+
+         restaurant = sample_restaurants.restaurants.find_one({"cuisine": "Bakery"})
+
+   .. tab:: Shell
+      :tabid: find-one-shell
+
+      .. code-block:: python
+         :copyable: true
+
+         >>> sample_restaurants.restaurants.find_one({"cuisine": "Bakery"})
+   
+.. tip:: Sort Order
+
+   The ``find_one()`` method returns the first document in 
+   :manual:`natural order </reference/glossary/#std-term-natural-order>`
+   on disk if no sort criteria is specified.
+
+.. To learn more about sorting, see the TODO: sorting page.
+
+.. TODO: To see a full example of using the ``find_one()`` method to find a single document, see
+   :ref:`pymongo-retrieve-additional-information`.
+
+.. _pymongo-retrieve-find-multiple:
+
+Find Multiple Documents
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To find multiple documents in a collection, pass a query filter to the ``find()``
+method that specifies the criteria of the documents you want to retrieve.
+
+The following examples use the ``find()`` method to find all documents where
+the ``"cuisine"`` field has the value ``"Spanish"``. Select the corresponding tab to see
+an example in the Python shell or a ``.py`` file. 
+
+.. tabs::
+
+   .. tab:: File
+      :tabid: find-multiple-file
+
+      .. code-block:: python
+         :copyable: true
+
+         cursor = sample_restaurants.restaurants.find({"cuisine": "Spanish"})
+
+   .. tab:: Shell
+      :tabid: find-multiple-shell
+
+      .. code-block:: python
+         :copyable: true
+
+         >>> sample_restaurants.restaurants.find({"cuisine": "Spanish"})
+
+You can use a **cursor** to iterate over the documents returned by the ``find()``
+method. A cursor is a mechanism that allows an application to iterate over database 
+results while holding only a subset of them in memory at a given time. Cursors
+are useful when your ``find()`` method returns a large amount of documents.
+
+You can iterate over the documents in a cursor by using a ``for-in`` loop, as shown in
+the following examples:
+
+.. tabs::
+
+   .. tab:: File
+      :tabid: iterate-file
+
+      .. code-block:: python
+         :copyable: true
+
+         cursor = sample_restaurants.restaurants.find({"cuisine": "Spanish"})
+         for restaurant in cursor:
```

Comment:
Might be useful to add an elipses in the code examples here so beginners know there needs to be more after this block. (Alternatively we could just show a print line or something) Applies to shell example too

```suggestion
         cursor = sample_restaurants.restaurants.find({"cuisine": "Spanish"})
         for restaurant in cursor:
         ...
```